app/components/CapacityBar.tsx

```diff
@@ -6,30 +6,41 @@
  * Copyright Oxide Computer Company
  */
 
-import { splitDecimal } from '@oxide/util'
+import cn from 'classnames'
+
+import { invariant, splitDecimal } from '@oxide/util'
 
 export const CapacityBar = ({
   icon,
   title,
   unit,
   provisioned,
   allocated,
-  allocatedLabel,
+  capacity,
   includeUnit = true,
 }: {
   icon: JSX.Element
   title: 'CPU' | 'Memory' | 'Storage'
   unit: 'nCPUs' | 'GiB' | 'TiB'
   provisioned: number
-  allocated: number
-  allocatedLabel: string
+  allocated?: number
+  capacity?: number
   includeUnit?: boolean
 }) => {
-  const percentOfAllocatedUsed = (provisioned / allocated) * 100
+  invariant(
+    typeof allocated === 'number' || typeof capacity === 'number',
+    'CapacityBar must be provided with at least one of allocated or capacity'
+  )
+  const hasAllocated = Number.isInteger(allocated)
+  const hasCapacity = Number.isInteger(capacity)
 
-  const [wholeNumber, decimal] = splitDecimal(percentOfAllocatedUsed)
+  const percentOfAllocatedUsed = (provisioned / (allocated || 0)) * 100
```

Comment:
Dividing by 0 seems like it might do something unexpected/unwanted? (next line as well)

Reply:
Weird thing w/ JS is that dividing by zero results in `Infinity`. I've leveraged that below (with `isFinite`) to make some logic distinctions. That's probably too clever by half and at least deserves an explanation via a comment if it stays.